src/main/java/ai/nettogrof/battlesnake/snakes/Snake.java

```diff
@@ -276,7 +276,7 @@ public Map<String, String> move(final JsonNode moveRequest) {
 			final String gameId = moveRequest.get("game").get("id").asText();
 			AbstractSnakeAI bot = bots.get(gameId);
 			if (bot == null) {
-				bots.put(gameId, new BetaSnake(gameId));
+				bots.put(gameId, new ExpertSnake(gameId));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.



**Problem**  
 You are using a `ConcurrentHashMap`, but your usage of `get()`, `null` check, and `put()` may not be thread-safe at lines: **277, 278, and 279**. Two threads can perform this **same check at the same time** and one thread can **overwrite** the value written by the other thread.

**Fix**  
 Consider replacing `put()` with `putIfAbsent()` to help prevent accidental overwriting. `putIfAbsent()` puts the value only if the `ConcurrentHashMap` does not contain the key and therefore avoids overwriting the value written there by the other thread&#39;s `putIfAbsent()`.

**More info**  
 `putIfAbsent()` returns `null` if the value did not exist and returns the value in the map if one already exists.

 [View an example on GitHub](https://github.com/apache/hadoop/blob/f16cf877e565084c66bc63605659b157c4394dc8/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/startupprogress/StartupProgress.java#L272-L273) (external link).